### PR TITLE
[acorn] 🌇 Contains built-in declarations

### PR DESCRIPTION
**Pull request by jablko**
These types appear to be for an [npm package](https://www.npmjs.com/package/acorn) that contains built-in TypeScript declarations as of version "7.1.1":

```sh
npm view acorn types
dist/acorn.d.ts
```

Can we retire our types now?

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

**Review by TeamworkGuy2 — APPROVED** [review]

File: types/detective/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "dependencies": {
+    "@types/acorn": "*"
```
Comment (peterblazejewicz) [inline types/detective/package.json:3]:
```suggestion
    "@types/acorn": "4.0.6"
```

File: types/detective/package.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "dependencies": {
+    "acorn": "^7.1.1"
```
Comment (peterblazejewicz) [inline types/detective/package.json:3]:
```suggestion
    "acorn": "^7.0.0"
```
https://github.com/browserify/acorn-node/blob/v1.8.2/package.json#L10
(as per 1.8.2, not 2.*)

File: types/detective/package.json (hunk 1 above)
Comment (peterblazejewicz) [reply types/detective/package.json]: ^7.0.0 was the latest version acorn version officially tested within native package
Yes, we can switch to native acorn on DT but for `detective` we should probably use same version as native package

**Review by RReverser — APPROVED** [review]
I haven't reviewed the actual changes, but overall retiring now that there are built-in types, sounds good.

**Pull request closed without merging**